apps/dashboard/src/components/pay/PayAnalytics/PayAnalytics.tsx

```diff
@@ -0,0 +1,214 @@
+import type { ApiKey } from "@3rdweb-sdk/react/hooks/useApi";
+import { format } from "date-fns";
+import { useState } from "react";
+import { DatePickerWithRange } from "../../../@/components/ui/DatePickerWithRange";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "../../../@/components/ui/select";
+import { PayCustomersTable } from "./components/PayCustomersTable";
+import { PayNewCustomers } from "./components/PayNewCustomers";
+import { PaymentHistory } from "./components/PaymentHistory";
+import { PaymentsSuccessRate } from "./components/PaymentsSuccessRate";
+import { Payouts } from "./components/Payouts";
+import { TotalPayVolume } from "./components/TotalPayVolume";
+import { TotalVolumePieChart } from "./components/TotalVolumePieChart";
+
+const durationPresets = [
+  {
+    name: "Last 7 Days",
+    id: "last-7",
+    days: 7,
+  },
+  {
+    name: "Last 30 Days",
+    id: "last-30",
+    days: 30,
+  },
+  {
+    name: "Last 60 Days",
+    id: "last-60",
+    days: 60,
+  },
+  {
+    name: "Last 120 Days",
+    id: "last-120",
+    days: 120,
+  },
+] as const;
+
+type DurationId = (typeof durationPresets)[number]["id"];
+
+type Range = {
+  type: DurationId | "custom";
+  label?: string;
+  from: Date;
+  to: Date;
+};
+
+export function PayAnalytics(props: { apiKey: ApiKey }) {
+  const clientId = props.apiKey.key;
+  const [range, setRange] = useState<Range>(() =>
+    getLastNDaysRange("last-120"),
+  );
+
+  const numberOfDays = Math.round(
```

Comment:
I see we're behaving _as if_ we already had react compiler :)

Reply:
It's a pretty lightweight calculation - don't really need a memo here